### PR TITLE
[CARBONDATA-1999] Block drop table and delete streaming segment while streaming is in progress

### PR DESCRIPTION
**Pull request by QiangCai**
1. Block drop table while streaming is in progress
2. Block delete streaming segment 

 - [x] Any interfaces changed?
 no
 - [x] Any backward compatibility impacted?
 no
 - [x] Document update required?
  yes
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
          added
        - How it is tested? Please attach test report.
          ut
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
small changes

### CODE REVIEW

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2785/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1389/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2624/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 1]
```diff
@@ -495,11 +495,16 @@ public static void writeLoadDetailsIntoFile(String dataLoadLocation,
           }
           // if the segment status is overwrite in progress, then no need to delete that.
           if (SegmentStatus.INSERT_OVERWRITE_IN_PROGRESS == loadMetadata.getSegmentStatus()) {
-            LOG.error("Cannot delete the segemnt " + loadId + " which is load overwrite " +
+            LOG.error("Cannot delete the segment " + loadId + " which is load overwrite " +
                     "in progress");
             invalidLoadIds.add(loadId);
             return invalidLoadIds;
           }
+          if (SegmentStatus.STREAMING == loadMetadata.getSegmentStatus()) {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: get the status once and use elseif

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 2]
```diff
@@ -543,6 +548,11 @@ public static void writeLoadDetailsIntoFile(String dataLoadLocation,
               + "as the segment has been compacted.");
           continue;
         }
+        if (SegmentStatus.STREAMING == loadMetadata.getSegmentStatus()) {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: get the status once and use elseif

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java [hunk 3]
```diff
@@ -543,6 +548,11 @@ public static void writeLoadDetailsIntoFile(String dataLoadLocation,
...
+          LOG.info("Ignoring the segment : " + loadMetadata.getLoadName()
+              + "as the segment is streaming in progress.");
+          continue;
+        }
         if (SegmentStatus.MARKED_FOR_DELETE != loadMetadata.getSegmentStatus() &&
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: get the status once and use elseif

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala [hunk 4]
```diff
@@ -55,6 +55,10 @@ case class CarbonDropTableCommand(
       }
       LOGGER.audit(s"Deleting table [$tableName] under database [$dbName]")
       carbonTable = CarbonEnv.getCarbonTable(databaseNameOp, tableName)(sparkSession)
+      if (carbonTable.isStreamingTable) {
+        // streaming table should acquire streaming.lock
+        carbonLocks += CarbonLockUtil.getLockObject(identifier, LockUsage.STREAMING_LOCK)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala]: better to assign to locksToBeAcquired at line 48 when initializing it.
```
   val locksToBeAcquired = if (streaming) {
       3 locks.
   } else {
       2 locks
  }
```

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java (hunk 1 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: fixed

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java (hunk 3 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: fixed

File: core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java (hunk 2 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java]: fixed

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala (hunk 4 above)
Comment (QiangCai) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala]: Here should get LockUsage.METADATA_LOCK before invoking  CarbonEnv.getCarbonTable to get metadata.

Comment (jackylk) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1394/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2629/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2791/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1403/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2636/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2797/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1407/

**Comment by jackylk** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2642/

**Pull request closed without merging**